### PR TITLE
[SP-4282] Backport of PDI-16921 - Concurrent access to outputRowSets …

### PR DESCRIPTION
**Pull request by pentaho-lmartins**
…in Mapping Init (8.0 Suite)

### CODE REVIEW

**Comment by pentaho-lmartins** [comment]
@ricardosilva88

**Comment by ricardosilva88** [comment]
@pentaho-lmartins please fix copyright year

File: engine/src/main/java/org/pentaho/di/trans/step/BaseStep.java [hunk 1]
```diff
@@ -2519,175 +2590,183 @@ public void dispatch() {
...
+            }
           }
         }
       }
+    } finally {
+      inputRowSetsLock.writeLock().unlock();
+      outputRowSetsLock.writeLock().unlock();
     }
 
     if ( stepMeta.getTargetStepPartitioningMeta() != null ) {
```
[the remark this answers is not in the document]
Comment (pentaho-lmartins) [reply engine/src/main/java/org/pentaho/di/trans/step/BaseStep.java]: It's here: https://github.com/pentaho-lmartins/pentaho-kettle/blob/2e4c4b02ac4699e1f5e46300e13eaecfb95e3134/engine/src/main/java/org/pentaho/di/trans/step/BaseStep.java#L2773

Same as in master. @ricardosilva88

[the remark this answers is not in the document]
Comment (pentaho-lmartins) [reply engine/src/main/java/org/pentaho/di/trans/step/BaseStep.java]: Updated copyrights. @ricardosilva88

**Comment by pentaho-lmartins** [comment]
Updated copyrights. @ricardosilva88

**Comment by ricardosilva88** [comment]
I'll accept the PR, since the build completed and broken tests aren't related with this PR

**Pull request merged**